File: PMAWGAN-Human3.6M/q_to_curve.py
import numpy as np
from numpy import linalg as LA
from scipy import integrate


def q_to_curves( q ):
    [L, F] = np.shape(q)
    s = np.linspace(0, 1, F)
    qnorm = np.zeros([F, 1])
    for i in range(F):
        qnorm[i] = LA.norm(q[:, i], 2)

    s = np.expand_dims(s, axis=0)
    curve = np.zeros([L, F])
    for i in range(L):
         temp = np.multiply(q[i,:],np.transpose(qnorm))
         curve[i, :] = integrate.cumtrapz(temp, s, initial=0)
    return curve

File: PMAWGAN-CMU/q_to_curve.m
function [ curve ] = q_to_curve( q )
%Q_TO_CURVE Summary of this function goes here
%   Detailed explanation goes here


[L,F]=size(q);
s=linspace(0,1,F);
qnorm=zeros(F,1);
for i=1:F
    qnorm(i)=norm(q(:,i));
end



for i=1:L
    temp=q(i,:).*qnorm';
    curve(i, :)=cumtrapz(s,temp);

end
% curve(:, end)
% size(curve)


end



